File: src/desktop/PhoneLink.Core/Models/TransferManifest.cs
namespace PhoneLink.Core.Models;

public enum TransferPurpose
{
    Photo = 1,
    Question = 2,
    File = 3,
}

public sealed record TransferManifest(
    string TransferId,
    string SenderDeviceId,
    string OriginalFileName,
    string MimeType,
    long FileSize,
    int? Width,
    int? Height,
    string Sha256,
    DateTimeOffset CapturedAt,
    DateTimeOffset SentAt,
    TransferPurpose Purpose);

File: src/android/PhoneLinkAndroid/app/src/main/java/com/phonelink/app/transfer/TransferManifest.kt
package com.phonelink.app.transfer

import org.json.JSONObject

/**
 * 上传 metadata（与 Phase 1 Windows 端 TransferManifest 契约一致，字段名小写固定）。
 */
data class TransferManifest(
    val transferId: String,
    val senderDeviceId: String,
    val originalFileName: String,
    val mimeType: String,
    val fileSize: Long,
    val width: Int,
    val height: Int,
    val sha256: String,
    val capturedAt: String,
    val sentAt: String,
    val purpose: String,
) {
    fun toJson(): String = JSONObject()
        .put("transferId", transferId)
        .put("senderDeviceId", senderDeviceId)
        .put("originalFileName", originalFileName)
        .put("mimeType", mimeType)
        .put("fileSize", fileSize)
        .put("width", width)
        .put("height", height)
        .put("sha256", sha256)
        .put("capturedAt", capturedAt)
        .put("sentAt", sentAt)
        .put("purpose", purpose)
        .toString()
}

object TransferIds {
    fun newTransferId(): String = "t-" + java.util.UUID.randomUUID().toString().replace("-", "")

    fun newSenderDeviceId(): String = "mobile-" + java.util.UUID.randomUUID().toString().replace("-", "")
}